pkg/controllers/deprovisioning/controller.go

```diff
@@ -106,8 +106,11 @@ func (c *Controller) Builder(_ context.Context, m manager.Manager) controller.Bu
 }
 
 func (c *Controller) Reconcile(ctx context.Context, _ reconcile.Request) (reconcile.Result, error) {
+	// We need to ensure that our internal cluster state mechanism is synced before we proceed
+	// with making any scheduling decision off of our state nodes. Otherwise, we have the potential to make
+	// a scheduling decision based on a smaller subset of nodes in our cluster state than actually exist.
 	if !c.cluster.Synced(ctx) {
-		return reconcile.Result{Requeue: true}, nil
```

Comment:
this was an immediate require, not an exponential backoff right?  An exponential backoff should only occur if an error is returned.

Reply:
No, a requeue like this in controller-runtime is treated as a requeue with backoff. The only difference between this and the error version is that it won't print an error if you just do a `Requeue: true`